cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -115,20 +153,21 @@ func watchOnce(etcdClient *etcd.Client, kubeClient *kclient.Client) {
 		case SetServices, AddService:
 			for i := range ev.Services {
 				s := &ev.Services[i]
-				name := buildNameString(s.Name, s.Namespace, *domain)
-				err := addDNS(name, s, etcdClient)
-				if err != nil {
-					log.Printf("Failed to add DNS for %s: %v", name, err)
+				inserter := &etcdRecordInserter{
```

Comment:
This is a good catch, but it seems like a complicated way to get what amounts to a function callback.  Could it be simplified to something like:

```
mutateOrDie(func() error { return addDNS(name, s, etcdClient) })
```

I am not sure I see the need for an interface and multiple implementations just to get one callback.